src/Compilers/CSharp/Portable/Symbols/NamedTypeSymbol.cs

```diff
@@ -1439,6 +1449,12 @@ ISymbol INamedTypeSymbol.AssociatedSymbol
         /// </summary>
         ImmutableArray<string> INamedTypeSymbol.TupleElementNames => this.TupleElementNames;
 
+        /// <summary>
+        /// If this is a tuple type symbol, returns the symbol for its underlying type.
+        /// Otherwise, returns null.
+        /// </summary>
+        INamedTypeSymbol INamedTypeSymbol.TupleUnderlyingType => this.TupleUnderlyingType;
+
```

Comment:
We have a few APIs that should only be called for tuple types. Would it make sense to assert (you shouldn't call those APIs for non-tuple types)?